xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/XMLUtils.java

```diff
@@ -218,37 +219,121 @@ public static String unescapeXMLComment(String content)
     }
 
     /**
-     * Escapes all the XML special characters in a <code>String</code> using numerical XML entities. Specifically,
-     * escapes &lt;, &gt;, ", ', &amp; and {.  Left curly bracket is included here to protect against {{/html}} in
-     * xwiki 2.x syntax.
+     * Escapes all the XML special characters and a XWiki Syntax 2.0+ special character (i.e., <code>{</code>, to
+     * protect against <code>{{/html}}</code>) in a {@code String}.
+     * The escaping is done using numerical XML entities to allow the content to be used as an XML attribute value
+     * or as an XML element text.
+     * For instance, {@code <b>{{html}}$x{{/html}}</b>} will be escaped and can thus be put inside an XML attribute.
+     * To illustrate, the value can be used in a div tag
+     * <code>&lt;div&gt;&amp;#60;b&amp;#62;&amp;#123;&amp;#123;html}}$x&amp;#123;&amp;#123;/html}}&amp;#60;/b&amp;#62;
+     * &lt;/div&gt;</code>
+     * or in the attribute of an input tag
+     * <code>&lt;input
+     * value=&quot;&amp;#60;b&amp;#62;&amp;#123;&amp;#123;html}}$x&amp;#123;&amp;#123;/html}}&amp;#60;/b&amp;#62;&quot;
+     * /&gt;</code>.
+     * <p>
+     * Specifically, escapes &lt;, &gt;, ", ', &amp; and {.
+     * <p>
+     * Note that is is preferable to use {@link #escapeAttributeValue(String)} when the content is used as
+     * an XML tag attribute, and {@link #escapeElementText(String)} when the content is used as an XML text.
      *
-     * @param content the text to escape, may be {@code null}
+     * @param content the text to escape, may be {@code null}. The content is converted to {@code String} using
+     * {@link Objects#toString(Object, String)}, where the second parameter is {@code null}
      * @return a new escaped {@code String}, {@code null} if {@code null} input
+     * @see #escapeAttributeValue(String)
+     * @see #escapeElementText(String)
+     * @deprecated since 12.8RC1, use {@link #escape(String)} instead
      */
+    @Deprecated
     public static String escape(Object content)
     {
-        return escapeAttributeValue(content);
+        return escapeAttributeValue(Objects.toString(content, null));
```

Comment:
I would call ``escape`` here to be in line with the ``@deprecated`` message and to make sure that both ``escape`` methods behave the same (someone may update ``escape(String)`` without updating the deprecated ``escape(Object)``). This is minor but it's a best practice IMO.

Reply:
I agree, `escape(Object)` is now calling `escape(String)` which itself calls `escapeAttributeValue(String)`.